### DISCUSSION TITLE
Help on migration

### DISCUSSION THREAD

**Discussion by sandbox-apps** [prisma/prisma#9218] [Q&A] [answered]
Hello prisma devs,

I am currently doing migration with the DATABASE_URL and SHADOW_DATABASE_URL. As far as I understand the docs when you use development commands like `npx prisma migrate dev` it will apply to SHADOW_DATABASE_URL? but in my case it still apply on the production url which is DATABASE_URL.. is this desired behaviour? or did I miss something?

**Comment by sandbox-apps** [comment]
Feel free to correct me if I misunderstood shadowing? Because I use it as for development iteration

**Reply by ryands17** [reply]
`prisma migrate dev` applies on the actual database. The shadow database is only used for helping with the migration. The actual migration is always applied on the main database.

**Reply by sandbox-apps** [reply]
How does shadow database differs from development database? i cant quite grasps what is shadow database and how can it help in migration

**Reply by ryands17** [reply, marked as answer]
Have a look at the introduction section of this doc where it is explained: https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
